layer/nrlf/core/validators.py

```diff
@@ -389,7 +444,30 @@ def _validate_category(self, model: DocumentReference):
                 issue_code="value",
                 error_code="INVALID_RESOURCE",
                 diagnostics=f"category code '{coding.code}' must have a display value of '{category_attributes.get('display')}'",
-                field=f"category[0].coding[{0}].display",
+                field="category[0].coding[0].display",
+            )
+
+    def _validate_type_category_mapping(self, model: DocumentReference):
+        """
+        Validate the type field matches the expected category
+        """
+        logger.log(LogReference.VALIDATOR001, step="type_category_mapping")
+
+        type_coding = model.type.coding[0]
+        type_id = f"{type_coding.system}|{type_coding.code}"
+        category_coding = model.category[0].coding[0]
+        category_id = f"{category_coding.system}|{category_coding.code}"
+
+        if type_id not in PointerTypes.list() or category_id not in Categories.list():
```

Comment:
would this ever even happen considering previous validation?

Reply:
Yes, it would happen, the previous validation is on another function that would exit early if there is an error and then come to this one.

Reply:
That already happens in the type/category validation, if they are not in that list, it would throw an error